src/components/Figure/Image.js

```diff
@@ -61,6 +64,10 @@ Image.propTypes = {
   aboveTheFold: PropTypes.bool
 }
 
+Image.contextTypes = {
+  toggleGallery: PropTypes.func
```

Comment:
Cool, I didn't know about the context feature in React. I assume it would be hard to pipe `toggleGallery` down to the component through the schema otherwise.

At first it looked a bit fishy to me to have the styleguide component depend on the context of the frontend parent component. But since it's optional, it's probably not that fishy. Would you see any other caveats from your experience?

Reply:
Prior to React v16.3, context used to be an "undocumented feature" to pass data through the component tree with out explicit props – the [new, official Context API](https://reactjs.org/docs/context.html) will be even nicer and easier to use.

I don't think having context provider and consumer in different projects is a problem – it works just like props, but behind the scenes. However, the need to use context is usually a hint that the design is flawed – but in this case I see no other way to distribute the global handler through the schema.